R/R/install.R

```diff
@@ -0,0 +1,150 @@
+#' Install Metaflow Python package
+#'
+#' This function wraps installation functions from [reticulate][reticulate::reticulate] to install the Python packages
+#' **metaflow** and it's Python dependencies.
+#'
+#' This package uses the [reticulate][reticulate::reticulate] package
+#' to make an interface with the [Metaflow](https://metaflow.org/)
+#' Python package.
+#'
+#' @param method `character`, indicates to use `"conda"` or `"virtualenv"`.
+#' @param version `character`, version of Metaflow to install. The default version
+#' is the latest available on PyPi.
+#' @param ... other arguments sent to [reticulate::conda_install()] or
+#'    [reticulate::virtualenv_install()]
+#'
+#' @seealso
+#' [reticulate: Using reticulate in an R Package](https://rstudio.github.io/reticulate/articles/package.html),
+#' [reticulate: Installing Python Packages](https://rstudio.github.io/reticulate/articles/python_packages.html)
+#' @examples
+#' \dontrun{
+#' # not run because it requires Python
+#' install_metaflow()
+#' }
+#' @export
+install_metaflow <- function(method = c("conda", "virtualenv"),
+                             version = NULL,
+                             ...) {
+  envname <- pkg.env$envname
+
+  env_set <- check_environment(envname)
+  if (method == "conda" && env_set[["virtualenv"]]){
+    message("Installing Metaflow python backend using conda but it has been installed in virtualenv.")
+    message("We do not allow python backend in both conda and virtualenv.")
+    message("1. To reinstall python backend in virtualenv: install_metaflow(method='virtualenv')")
+    message("2. To remove python backend: remove_metaflow()")
```

Comment:
I am not sure it follows immediately, in what scenario they are supposed to invoke `remove_metaflow()`. If it is an edge case, then we should just mention that in our docs instead and remove it from here.